### ISSUE TITLE
Can't place JTable

### ISSUE THREAD

**Issue by riwarek** [eclipse-windowbuilder/windowbuilder#588]
When I try to place JTabel in WindowsBuilder by clicking on JTable component nothing happen. In log I see what below.
When I define and declare JTable someTable = new JTable() in Source view, after retrun to Design view whole area is white.
Eclipse 2023-06, 
WindowBuilder components 1.13.0.202309050414	org.eclipse.wb.core.feature.feature.group	Eclipse WindowBuilder
Java 21

Palette: can not load component javax.swing.JTable

```
Stack trace:
java.lang.ClassNotFoundException: java.lang.Compiler
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:446)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:416)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at org.mvel2.compiler.AbstractParser.setupParser(AbstractParser.java:215)
	at org.mvel2.compiler.AbstractParser.<clinit>(AbstractParser.java:153)
	at org.mvel2.MVEL.compileExpression(MVEL.java:827)
	at org.eclipse.wb.internal.core.model.util.ScriptUtils.compile(ScriptUtils.java:230)
	at org.eclipse.wb.internal.core.model.util.ScriptUtils.evaluate(ScriptUtils.java:85)
	at org.eclipse.wb.internal.core.model.util.ScriptUtils.evaluate(ScriptUtils.java:47)
	at org.eclipse.wb.internal.core.model.description.rules.PropertyDefaultRule.getValue(PropertyDefaultRule.java:60)
	at org.eclipse.wb.internal.core.model.description.rules.PropertyDefaultRule.begin(PropertyDefaultRule.java:48)
	at org.apache.commons.digester3.Digester.startElement(Digester.java:1350)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:518)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanStartElement(XMLDocumentFragmentScannerImpl.java:1387)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2726)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
	at org.apache.commons.digester3.Digester.parse(Digester.java:1642)
	at org.apache.commons.digester3.Digester.parse(Digester.java:1701)
	at org.eclipse.wb.internal.core.model.description.helpers.ComponentDescriptionHelper.getDescription0(ComponentDescriptionHelper.java:375)
	at org.eclipse.wb.internal.core.model.description.helpers.ComponentDescriptionHelper.getDescription0(ComponentDescriptionHelper.java:280)
	at org.eclipse.wb.internal.core.model.description.helpers.ComponentDescriptionHelper.getDescription(ComponentDescriptionHelper.java:249)
	at org.eclipse.wb.core.editor.palette.model.entry.ComponentEntryInfo.ensureDescriptions(ComponentEntryInfo.java:279)
	at org.eclipse.wb.core.editor.palette.model.entry.ComponentEntryInfo.createTool(ComponentEntryInfo.java:411)
	at org.eclipse.wb.core.editor.palette.model.entry.ToolEntryInfo$1.runObject(ToolEntryInfo.java:39)
	at org.eclipse.wb.core.editor.palette.model.entry.ToolEntryInfo$1.runObject(ToolEntryInfo.java:1)
	at org.eclipse.wb.internal.core.utils.execution.ExecutionUtils.runObjectLog(ExecutionUtils.java:311)
	at org.eclipse.wb.core.editor.palette.model.entry.ToolEntryInfo.activate(ToolEntryInfo.java:33)
	at org.eclipse.wb.internal.core.editor.palette.DesignerPalette$3.activate(DesignerPalette.java:301)
	at org.eclipse.wb.core.controls.palette.PaletteComposite.selectEntry(PaletteComposite.java:241)
	at org.eclipse.wb.core.controls.palette.PaletteComposite$EntryFigure$1.mouseReleased(PaletteComposite.java:786)
	at org.eclipse.draw2d.Figure.handleMouseReleased(Figure.java:989)
	at org.eclipse.wb.internal.draw2d.EventManager.lambda$4(EventManager.java:176)
	at org.eclipse.wb.internal.draw2d.EventManager.delayEvent(EventManager.java:255)
	at org.eclipse.wb.internal.draw2d.EventManager.sendEvent(EventManager.java:202)
	at org.eclipse.wb.internal.draw2d.EventManager.mouseUp(EventManager.java:176)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:224)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4274)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4072)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3660)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:643)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:550)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)


Full stack trace (to see full context):
java.lang.NoClassDefFoundError: java/lang/Compiler
	at org.mvel2.compiler.AbstractParser.setupParser(AbstractParser.java:215)
	at org.mvel2.compiler.AbstractParser.<clinit>(AbstractParser.java:153)
	at org.mvel2.MVEL.compileExpression(MVEL.java:827)
	at org.eclipse.wb.internal.core.model.util.ScriptUtils.compile(ScriptUtils.java:230)
	at org.eclipse.wb.internal.core.model.util.ScriptUtils.evaluate(ScriptUtils.java:85)
	at org.eclipse.wb.internal.core.model.util.ScriptUtils.evaluate(ScriptUtils.java:47)
	at org.eclipse.wb.internal.core.model.description.rules.PropertyDefaultRule.getValue(PropertyDefaultRule.java:60)
	at org.eclipse.wb.internal.core.model.description.rules.PropertyDefaultRule.begin(PropertyDefaultRule.java:48)
	at org.apache.commons.digester3.Digester.startElement(Digester.java:1350)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:518)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanStartElement(XMLDocumentFragmentScannerImpl.java:1387)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2726)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:542)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:889)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:825)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1224)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
	at org.apache.commons.digester3.Digester.parse(Digester.java:1642)
	at org.apache.commons.digester3.Digester.parse(Digester.java:1701)
	at org.eclipse.wb.internal.core.model.description.helpers.ComponentDescriptionHelper.getDescription0(ComponentDescriptionHelper.java:375)
	at org.eclipse.wb.internal.core.model.description.helpers.ComponentDescriptionHelper.getDescription0(ComponentDescriptionHelper.java:280)
	at org.eclipse.wb.internal.core.model.description.helpers.ComponentDescriptionHelper.getDescription(ComponentDescriptionHelper.java:249)
	at org.eclipse.wb.core.editor.palette.model.entry.ComponentEntryInfo.ensureDescriptions(ComponentEntryInfo.java:279)
	at org.eclipse.wb.core.editor.palette.model.entry.ComponentEntryInfo.createTool(ComponentEntryInfo.java:411)
	at org.eclipse.wb.core.editor.palette.model.entry.ToolEntryInfo$1.runObject(ToolEntryInfo.java:39)
	at org.eclipse.wb.core.editor.palette.model.entry.ToolEntryInfo$1.runObject(ToolEntryInfo.java:1)
	at org.eclipse.wb.internal.core.utils.execution.ExecutionUtils.runObjectLog(ExecutionUtils.java:311)
	at org.eclipse.wb.core.editor.palette.model.entry.ToolEntryInfo.activate(ToolEntryInfo.java:33)
	at org.eclipse.wb.internal.core.editor.palette.DesignerPalette$3.activate(DesignerPalette.java:301)
	at org.eclipse.wb.core.controls.palette.PaletteComposite.selectEntry(PaletteComposite.java:241)
	at org.eclipse.wb.core.controls.palette.PaletteComposite$EntryFigure$1.mouseReleased(PaletteComposite.java:786)
	at org.eclipse.draw2d.Figure.handleMouseReleased(Figure.java:989)
	at org.eclipse.wb.internal.draw2d.EventManager.lambda$4(EventManager.java:176)
	at org.eclipse.wb.internal.draw2d.EventManager.delayEvent(EventManager.java:255)
	at org.eclipse.wb.internal.draw2d.EventManager.sendEvent(EventManager.java:202)
	at org.eclipse.wb.internal.draw2d.EventManager.mouseUp(EventManager.java:176)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:224)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4274)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1066)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4072)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3660)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:643)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:550)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:171)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)
Caused by: java.lang.ClassNotFoundException: java.lang.Compiler
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:446)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:416)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:168)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 65 more

```

**Comment by merks** [comment]
That class is marked for removal in Java 17.  Which JDK are you using?  

WindowBuilder is using the newest version of that library so this is not easily fixed with some update...

https://repo1.maven.org/maven2/org/mvel/mvel2/2.5.0.Final/

**Comment by riwarek** [comment]
java 21 2023-09-19 LTS

**Comment by riwarek** [comment]
Workaround for this is running Eclipse with:

-vm
c:\Program Files\Java\jdk-17\bin\javaw

in eclipse.ini

**Comment by ptziegler** [comment]
This is a duplicate of #579 and has already been fixed with #581.

In short: The MVEL version distributed with current release (2.4.15) is not compatible with Java 21. So your options are to:
- Stay to Java 17 until the winter release
- Manually download and install MVEL 2.5.0
- Use the 1.14.0 M1: https://download.eclipse.org/windowbuilder/updates/milestone/S20231002053

> That class is marked for removal in Java 17. Which JDK are you using?
> WindowBuilder is using the newest version of that library so this is not easily fixed with some update...

Only with the upcoming 1.14.0. The current release is still using the incompatible version.

> Do I understand correctly this is internal problem of WindowBuilder because of changes in JDK?
> But why only in case of JTable? Rest of components work properly.

MVEL is only used sporadically to evaluate the Java code. It's much more noticable with the SWT components.
